tests/unit/backend/native/test_engine.py

```diff
@@ -0,0 +1,325 @@
+from otx.types.export import OTXExportFormatType
+from otx.types.label import NullLabelInfo
+from otx.types.precision import OTXPrecisionType
+
+
+@pytest.fixture()
+def fxt_engine(tmp_path) -> OTXEngine:
+    return OTXEngine(
+        data="tests/assets/classification_dataset",
+        task="MULTI_CLASS_CLS",
+        model="tv_mobilenet_v3_small",
+        work_dir=tmp_path,
+        max_epochs=90,
+    )
+
+
+class TestEngine:
+    def test_constructor(self, tmp_path) -> None:
+        with pytest.raises(RuntimeError):
+            OTXEngine(work_dir=tmp_path)
+
+        # Check auto-configuration
+        data_root = "tests/assets/classification_dataset"
+        engine = OTXEngine(work_dir=tmp_path, data=data_root)
+        assert engine.task == "MULTI_CLASS_CLS"
+        assert engine.datamodule.task == "MULTI_CLASS_CLS"
+        assert isinstance(engine.model, EfficientNetMulticlassCls)
+
+        assert "default_root_dir" in engine.trainer_params
+        assert engine.trainer_params["default_root_dir"] == tmp_path
+        assert "accelerator" in engine.trainer_params
+        assert engine.trainer_params["accelerator"] == "auto"
+        assert "devices" in engine.trainer_params
+        assert engine.trainer_params["devices"] == 1
+
+        # Create engine with no data_root
+        with pytest.raises(ValueError, match="Given model class (.*) requires a valid label_info to instantiate."):
+            _ = OTXEngine(work_dir=tmp_path, task="MULTI_CLASS_CLS")
+
+    @pytest.fixture()
+    def mock_datamodule(self, mocker):
+        mock_datamodule = MagicMock()
+        mock_datamodule.label_info = 4321
+        mock_datamodule.input_size = (1234, 1234)
+        mock_datamodule.input_mean = (0.0, 0.0, 0.0)
+        mock_datamodule.input_std = (1.0, 1.0, 1.0)
+
+        return mocker.patch(
+            "otx.tools.auto_configurator.AutoConfigurator.get_datamodule",
+            return_value=mock_datamodule,
+        )
+
+    def test_model_init(self, tmp_path, mock_datamodule):
+        data_root = "tests/assets/classification_dataset"
+        engine = OTXEngine(work_dir=tmp_path, data=data_root)
+
+        assert engine._model.data_input_params == DataInputParams((1234, 1234), (0.0, 0.0, 0.0), (1.0, 1.0, 1.0))
+        assert engine._model.label_info.num_classes == 4321
+
+    def test_model_setter(self, fxt_engine, mocker) -> None:
+        assert isinstance(fxt_engine.model, TVModelMulticlassCls)
+        fxt_engine.model = "efficientnet_b0"
+        assert isinstance(fxt_engine.model, EfficientNetMulticlassCls)
+
+    def test_training_with_override_args(self, fxt_engine, mocker) -> None:
+        mocker.patch("pathlib.Path.symlink_to")
+        mocker.patch("otx.backend.native.engine.Trainer.fit")
+        mock_seed_everything = mocker.patch("otx.backend.native.engine.seed_everything")
+
+        assert fxt_engine._cache.args["max_epochs"] == 90
+
+        fxt_engine.train(max_epochs=100, seed=1234)
```

Comment:
Is it necessary to train for such a high number of epochs (e.g., 90 or 100)? Could a smaller number suffice for unit test (e.g 5-10)?